src/Apps/Order/Routes/Review/index.tsx

```diff
@@ -346,6 +351,17 @@ export const ReviewRoute: FC<ReviewProps> = props => {
             title: title,
             message: message,
           })
+        } else if (data.decline_code === "currency_not_supported") {
+          const { title, message, formattedMessage } = getErrorDialogCopy(
+            ErrorDialogs.CurrencyNotSupported
+          )
+
+          trackErrorMessageEvent(title, message, data.decline_code)
+
+          await props.dialog.showErrorDialog({
+            title: title,
+            message: formattedMessage,
+          })
```

Comment:
can this logic be moved into an `<ErrorBoundary />`, with exchange mutations receiving an error type just converting and emitting that up?